### DISCUSSION TITLE
Help to re-encode RTMP or RTSP stream to a new RTSP stream

### DISCUSSION THREAD

**Discussion by PedroCPereiraDev** [bluenviron/mediamtx#848] [Questions]
Hello,

I am loosing my sanity trying to set up a way to re-encode a RTSP or RTMP stream (tried both) to h264 to send in a RTSP stream.
I read all the documentation and did many experimentations.
I tried also running the ffmpeg commands directly on my computer and although I do not see any error my VLC cannot open the stream.

An exemple of my latest tries, all failed:

> paths:
>   all:
>   cam/realmonitor:
>     runOnDemand: ffmpeg -i rtmp://192.168.0.93/bcs/channel0_ext.bcs?channel=0&stream=0&user=admin&password=password -c:v libx264 -preset ultrafast -b:v 500k -max_muxing_queue_size 1024 -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
>     runOnDemandRestart: yes
> 
>   banana:
>     runOnDemand: ffmpeg -i "rtmp://192.168.0.93/bcs/channel0_ext.bcs?channel=0&stream=0&user=admin&password=password" -c copy -f rtsp "rtsp://localhost:8554/banana"
>     runOnDemandRestart: yes

    
When I run rtsp-simple-server docker and try to acess the streams the error appears as such: `runOnDemand command exited with code 0`

**Comment by aler9** [comment]
From the README:

> Please keep in mind that the Docker image doesn't include FFmpeg. if you need to use FFmpeg for an external command or anything else, you need to build a Docker image that contains both rtsp-simple-server and FFmpeg, by following instructions [here](https://github.com/aler9/rtsp-simple-server/discussions/278#discussioncomment-549104).

[a comment by an automated account is not shown]
